Services/CovidService.cs

```diff
@@ -17,16 +18,33 @@ public CovidService(IConfiguration config)
             _baseUrl = config["Covid19ApiUrl"];
         }
 
-        public async Task<List<CovidStatus>> GetStatus()
+        /// <summary>
+        /// Gets the status asynchronous.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IEnumerable<CovidStatus>> GetStatusAsync()
         {
-            return await _baseUrl
-                .AppendPathSegment("status")
-                .GetJsonAsync<List<CovidStatus>>();
+            var url = _baseUrl.AppendPathSegment(RouteConstants.ByStatus);
+            var result = await url.GetJsonAsync<List<CovidStatus>>().ConfigureAwait(false);
+
+            return result;
         }
 
-        public Task<List<CovidStatus>> GetStatusByCountry(string countryCode)
+        /// <summary>
+        /// Gets the status by country asynchronous.
+        /// </summary>
+        /// <param name="countryCode">The country code.</param>
+        /// <returns></returns>
+        public async Task<IEnumerable<CovidStatus>> GetStatusByCountryAsync(string countryCode)
         {
-            throw new System.NotImplementedException();
+            var query = await _baseUrl
+                .AppendPathSegment(RouteConstants.ByStatus)
+                .GetJsonAsync<List<CovidStatus>>()
+                .ConfigureAwait(false);
+
+            query = query.Where(p => p.Country.Equals(countryCode)).ToList();
```

Comment:
Should we care about casing in this comparison?

Reply:
good question , let me take a look at the api

Reply:
I just updated the new change, by the way good question